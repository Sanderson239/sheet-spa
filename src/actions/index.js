export const SHEET_RETRIEVED = 'SHEET_RETRIEVED';
/* eslint-disable no-debugger */

export function fetchSheet(spreadsheetId, sheetName) {
  const queryString =
  `{
      sheets(spreadsheetId:"${spreadsheetId}", sheetName:"${sheetName}") {
        spreadsheetId
        sheets {
          sheetName
          majorDimension
          range
          values
        }
      }
    }`;
  const request = { query: queryString };
  return (dispatch, getState, { SHEETS_API }) => {
    SHEETS_API.fetchData(request)
      .then(response => response.data)
      .then((sheet) => {
        dispatch({
          type: SHEET_RETRIEVED,
          sheetData: sheet,
        });
      });
  };
}
