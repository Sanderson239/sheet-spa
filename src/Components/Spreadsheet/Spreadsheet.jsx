import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Table } from 'semantic-ui-react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { fetchSheet } from '../../actions';
import { HeaderCell } from '../HeaderCell/HeaderCell';
import { Row } from '../Row/Row';
import injectWidgetId from '../../utils/utils';


export class Spreadsheet extends Component {
  componentDidMount() {
    this.props.fetchSheet('1UcfQsQGTAAtjvyxv948z3hf0qiUnMNZF90-GcD7MF9g', 'Sheet1');
  }
  /* eslint-disable max-len */


  render() {
    const headerCellIds = this.props.sheet.header.headerCellIds;
    const rowIds = this.props.sheet.rowIds;

    return (
      <div className="spreadsheet">
        <Table celled striped>
          <Table.Header>
            <Table.Row>
              { headerCellIds.map(headerCellId => <HeaderCell key={headerCellId} headerCellId={headerCellId} />) }
            </Table.Row>
          </Table.Header>
          <Table.Body>
            { rowIds.map(rowId => <Row key={rowId} rowId={rowId} />) }
          </Table.Body>
        </Table>
      </div>
    );
  }
}

Spreadsheet.propTypes = {
  fetchSheet: PropTypes.func.isRequired,
  sheet: PropTypes.shape({ range: PropTypes.string, majorDimension: PropTypes.string, header: PropTypes.shape({ headerCellIds: PropTypes.array, headerCellsById: PropTypes.shape({}) }), rowIds: PropTypes.array, rowsById: PropTypes.shape({ cellIds: PropTypes.array, cellsById: PropTypes.shape({}) }) }).isRequired,
};

const mapStateToProps = (state, ownProps) => {
  const id = ownProps.widgetId;
  const sheet = state.widgets.byId[id];
  return {
    sheet,
  };
};

export const mapDispatchToProps = dispatch => bindActionCreators({
  fetchSheet,
}, dispatch);

export default injectWidgetId(connect(mapStateToProps, mapDispatchToProps)(Spreadsheet));
