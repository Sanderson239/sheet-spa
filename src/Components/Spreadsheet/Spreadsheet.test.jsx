import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Spreadsheet, mapDispatchToProps } from './Spreadsheet';
import { fetchSheet } from '../../actions';

describe('Spreadsheet', () => {
  it('should render a Spreadsheet with range, majorDimension, header, rowIds, and rowsById', () => {
    const sheet = {
      header: { headerCellIds: [1, 2, 3, 4, 5], headerCellsById: { 1: 'AcolumnName', 2: 'BcolumnName', 3: 'CcolumnName', 4: 'DcolumnName', 5: 'EcolumnName' } },
      majorDimension: 'ROWS',
      range: 'Sheet1!A1:D5',
      rowIds: [1, 2, 3, 4],
      rowsById: {
        1: { cellIds: [1, 2, 3, 4, 5], cellsById: { 1: 'A2', 2: 'B2', 3: 'C2', 4: 'D2', 5: 'E2' } },
        2: { CellIds: [1, 2, 3, 4, 5], cellsById: { 1: 'A3', 2: 'B3', 3: 'C3', 4: 'D3', 5: 'E3' } },
        3: { CellIds: [1, 2, 3, 4, 5], cellsById: { 1: 'A4', 2: 'B4', 3: 'C4', 4: 'D4', 5: 'E4' } },
        4: { CellIds: [1, 2, 3, 4, 5], cellsById: { 1: 'A5', 2: 'B5', 3: 'C5', 4: 'D5', 5: 'E5' } },
      },
    };
    const component = shallow(
      <Spreadsheet
        sheet={sheet}
        fetchSheet={fetchSheet}
      />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  it('mapDispatchToProps', () => {
    const dispatch = jest.fn();

    expect(mapDispatchToProps(dispatch)).toHaveProperty('fetchSheet');
  });
});
