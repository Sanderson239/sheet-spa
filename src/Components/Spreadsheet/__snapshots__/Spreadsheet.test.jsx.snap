// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spreadsheet should render a Spreadsheet with range, majorDimension, header, rowIds, and rowsById 1`] = `
<div
  className="spreadsheet"
>
  <Table
    as="table"
    celled={true}
    striped={true}
  >
    <TableHeader
      as="thead"
    >
      <TableRow
        as="tr"
        cellAs="td"
      >
        <Wrapper
          headerCellId={1}
          key="1"
        />
        <Wrapper
          headerCellId={2}
          key="2"
        />
        <Wrapper
          headerCellId={3}
          key="3"
        />
        <Wrapper
          headerCellId={4}
          key="4"
        />
        <Wrapper
          headerCellId={5}
          key="5"
        />
      </TableRow>
    </TableHeader>
    <TableBody
      as="tbody"
    >
      <Connect(Component)
        key="1"
        rowId={1}
      />
      <Connect(Component)
        key="2"
        rowId={2}
      />
      <Connect(Component)
        key="3"
        rowId={3}
      />
      <Connect(Component)
        key="4"
        rowId={4}
      />
    </TableBody>
  </Table>
</div>
`;
